auth_test.go

```diff
@@ -8,30 +8,58 @@ import (
 
 // See helper_for_test.go for common infrastructure and tools.
 
-func (ts *TestSuite) Test_login_no_task_id() {
+// Can't log in without a taskID.
+func (ts *TestSuite) Test_login_no_taskID() {
 	ts.SetupBackend()
 	req := ts.mkReq("login", jsonobj{})
+	ts.HandleRequestError(req, "permission denied")
+}
+
+// Can't log in with a taskID that doesn't exist.
+func (ts *TestSuite) Test_login_missing_taskID() {
+	ts.SetupBackend()
+	req := ts.mkReq("login", jsonobj{"task-id": "missing-task.abc-123"})
+	ts.HandleRequestError(req, "permission denied")
+}
+
+// Can't log in with a taskID that doesn't have a Marathon-style app prefix.
+func (ts *TestSuite) Test_login_taskID_with_no_prefix() {
+	ts.SetupBackend()
+	temporarySetOfExistingTasks["abc-123"] = true
+	req := ts.mkReq("login", jsonobj{"task-id": "abc-123"})
+	ts.HandleRequestError(req, "permission denied")
+}
 
-	resp, err := ts.HandleRequest(req)
-	ts.EqualError(err, "permission denied")
-	ts.Nil(resp)
+// Can't log in with a taskID that doesn't have policies configured for its
+// prefix.
+func (ts *TestSuite) Test_login_unregistered_taskID() {
+	ts.SetupBackend()
+	temporarySetOfExistingTasks["unregistered-task.abc-123"] = true
+	req := ts.mkReq("login", jsonobj{"task-id": "unregistered-task.abc-123"})
+	ts.HandleRequestError(req, "permission denied")
 }
 
-func (ts *TestSuite) Test_login_good_task_id() {
+// Can log in with a taskID that exists and has policies configured for its
+// prefix.
+func (ts *TestSuite) Test_login_good_taskID() {
 	ts.SetupBackend()
-	req := ts.mkReq("login", jsonobj{"task-id": "task-that-exists"})
+	temporarySetOfExistingTasks["task-that-exists.abc-123"] = true
```

Comment:
This variable scoping 🤔

Reply:
That's a module-global private variable. If it were anything other than a temporary hack, I'd wrap it in helper functions or something to make it less magical.

Reply:
Do the values not need to be cleared at the end of a test then?

Reply:
We clear it in `TestSuite.SetupTest()` -- more magical poking at global state from multiple places. :-(